src/openforms/authentication/contrib/digid_eherkenning_oidc/plugin.py

```diff
@@ -31,30 +42,41 @@
 )
 
 
+class AuthInit(Protocol):
+    def __call__(
+        self, request: HttpRequest, return_url: str, *args, **kwargs
+    ) -> HttpResponseBase: ...
+
+
 class OIDCAuthentication(BasePlugin):
     verbose_name = ""
     provides_auth = ""
-    init_url = ""
     session_key = ""
     config_class = None
+    init_view: ClassVar[AuthInit]
 
     def start_login(self, request: HttpRequest, form: Form, form_url: str):
-        login_url = reverse(self.init_url, request=request)
-
         auth_return_url = reverse(
             "authentication:return",
             kwargs={"slug": form.slug, "plugin_id": self.identifier},
         )
-        return_url = furl(auth_return_url).set(
-            {
-                "next": form_url,
-            }
-        )
+        return_url = furl(auth_return_url).set({"next": form_url})
         if co_sign_param := request.GET.get(CO_SIGN_PARAMETER):
             return_url.args[CO_SIGN_PARAMETER] = co_sign_param
 
-        redirect_url = furl(login_url).set({"next": str(return_url)})
-        return HttpResponseRedirect(str(redirect_url))
+        # "evaluate" the view, this achieves two things:
+        #
+        # * we save a browser redirect cycle since we get the redirect to the identity
+        #   provider immediately
+        # * we control the config to apply 100% server side rather than passing it as
+        #   a query parameter, which prevents a malicious user from messing with the
+        #   redirect URL
+        #
+        # This may raise `OIDCProviderOutage`, which bubbles into the generic auth
+        # start_view and gets handled there.
+        response = self.init_view(request, return_url=str(return_url))
```

Comment:
this is not the most elegant thing, because it actually goes through the `dispatch` and `request.method` causes `.get()` to be invoked :sweat_smile: 

But basically `ClassBasedView.as_view()` turns the CBV into a "function based" view, or just view in this case because a view is nothing more than a callable taking a `request` instance which produces a `response`.

The nice thing is that you can use `as_view(**kwargs)` to pre-configure some aspects of the view (you can also do a similar thing at the URL level iirc, something like `path("...", CVB.as_view(), kwargs=some_kwargs)`

In this case, we don't want to hook up each view to it's own route, we just want its behaviour (since it is basically already hooked up via plugin-specific routes that already exist anyway).